### DISCUSSION TITLE
Datagrid is very slow when loading 500 records

### DISCUSSION THREAD

**Discussion by upswing1** [enisn/UraniumUI#232] [Q&A]
What is the fastest recommended method to load data in the datagrid? Currently I bind a collection of records but the rendering is very slow. Can the rendering be done only after all the data is loaded in the control?

**Comment by enisn** [comment]
It's discussed here: https://github.com/enisn/UraniumUI/issues/190#issuecomment-1374782449

@YBTopaz8 investigated issue and it seems it's not realted to UraniumUI. Maui has some performance issues on windows. Do you have same problem on other platforms too?

**Reply by enisn** [reply]
Also you can try [this workaround](https://stackoverflow.com/questions/13302933/how-to-avoid-firing-observablecollection-collectionchanged-multiple-times-when-r/13303245#13303245), it seems it's working better than current situation

**Reply by YBTopaz8** [reply]
I'm doing to side with you on this one!
It seems to be a MAUI issue on windows indeed.

I haven't had this problem on android AT ALL

**Reply by upswing1** [reply]
I have not tested on other platforms

Here is some data that will explain the problem:
loading first collection  (first time)
00:00:08.0762845
loading a second collection  (first time)
00:01:01.0588048
loading first collection  (second time)
00:02:35.0036999

So the same collection loading time goes from 8 seconds  to 2:35 minutes

I tried a different test 
loading first collection  (first time)
00:00:07.9608214
reloading immediately the first collection )  
00:00:01.4920819

loading second collection  (first time)
00:01:00.2177744
reloading immediately the second collection ) 
00:00:01.5378519

loading first collection  (second time)
00:02:34.9020166
reloading immediately the first collection )  
00:00:01.7811220

here is the test code for immediate reload

    private Stopwatch sw = new Stopwatch();
    private async void ViewDataButton_Clicked(object sender, EventArgs e)
    {

        bool answer = await Shell.Current.DisplayAlert("Warning", "Depending on the size of your data set this can take a while. Do you want to proceed?", "Yes", "No");
        BrowseButton.IsEnabled = true; progressBar.IsVisible = false;
        try
        {
            if (answer)
            {
                await LoadData();
                Thread.Sleep(1000);
                await LoadData();
            }
        }
        catch(Exception exc)
        {
            string s = exc.Message;
        }
            
        return;
    }
    async Task LoadData()
    {
        sw.Reset();
        sw.Start();

        Lbl_ViewData.IsVisible = true;
        ViewDataActivity.IsRunning = true;

        await Task.Run(() =>
        {
            Thread.Sleep(1000);
            MainThread.InvokeOnMainThreadAsync(() => DataGrid.ItemsSource = tempTasks);
        });

        sw.Stop();
        Debug.WriteLine(sw.Elapsed.ToString());
        progressBar.Progress = 0.8;

        ViewDataActivity.IsRunning = false;
        Lbl_ViewData.IsVisible = false;
  }

What can cause this behaviour in my code?
